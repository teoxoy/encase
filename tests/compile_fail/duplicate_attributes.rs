use encase::ShaderType;

fn main() {}

#[derive(ShaderType)]
struct Test {
    #[shader(align(8), align(16))]
    a: u32,
    #[shader(size(runtime), size(16))]
    b: u32,
}
