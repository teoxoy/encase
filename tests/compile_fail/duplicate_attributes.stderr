error: duplicate `align(X)` attribute
 --> tests/compile_fail/duplicate_attributes.rs:7:30
  |
7 |     #[shader(align(8), align(16))]
  |                              ^^

error: duplicate `size(X)` attribute
 --> tests/compile_fail/duplicate_attributes.rs:9:34
  |
9 |     #[shader(size(runtime), size(16))]
  |                                  ^^
